src/core/surface/completion_queue.c

```diff
@@ -283,6 +289,7 @@ grpc_event grpc_completion_queue_pluck(grpc_completion_queue *cc, void *tag,
   ret = ev->base;
   gpr_free(ev);
   GRPC_SURFACE_TRACE_RETURNED_EVENT(cc, &ret);
```

Comment:
have an exit label here maybe and use goto from everywhere else instead of duplicating the exit logic ?